pkg/reckoner/validation_test.go

```diff
@@ -39,76 +39,124 @@ func TestValidateArgs(t *testing.T) {
+				runAll:  true,
+				onlyRun: []string{"rbac-manager"},
 			},
-			want:    "",
 			wantErr: true,
 		},
 		{
-			name: "course.yaml not exist, but no -a or -o passed",
+			name: "pass onlyrun with success",
 			args: args{
-				args: []string{"testdata/course.yaml"},
+				args:    []string{},
+				runAll:  false,
+				onlyRun: []string{"rbac-manager"},
 			},
-			want:    "",
-			wantErr: true,
+			want:    "testdata/course.yaml",
+			wantErr: false,
 		},
 		{
-			name: "course.yaml not exist, but no -a or -o passed",
+			name: "pass runall with success",
 			args: args{
-				args: []string{"testdata/course.yaml"},
+				args:   []string{},
+				runAll: true,
 			},
-			want:    "",
-			wantErr: true,
+			want:    "testdata/course.yaml",
+			wantErr: false,
 		},
 		{
-			name: "course.yaml exists, Only one of runAll or onlyRun can be set",
+			name: "length of args = 2",
 			args: args{
-				args:    []string{"testdata/course.yaml"},
-				runAll:  true,
+				args:    []string{"testdata/course.yaml", "course.yml"},
+				runAll:  false,
 				onlyRun: []string{"rbac-manager"},
 			},
 			wantErr: true,
 		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := ValidateArgs(tt.args.runAll, tt.args.onlyRun, tt.args.args)
+			if tt.wantErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestValidateCourseFilePath(t *testing.T) {
+	type args struct {
+		runAll  bool
+		onlyRun []string
+		args    []string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    string
+		wantErr bool
+	}{
 		{
-			name: "course.yaml exists, pass onlyrun with success",
-			args: args{
-				args:    []string{"testdata/course.yaml"},
-				runAll:  false,
-				onlyRun: []string{"rbac-manager"},
-			},
-			want:    "testdata/course.yaml",
-			wantErr: false,
+			name:    "empty args",
+			args:    args{},
+			want:    "",
+			wantErr: true,
 		},
 		{
-			name: "course.yaml exists, pass runall with success",
+			name: "course.yaml does not exist",
 			args: args{
-				args:   []string{"testdata/course.yaml"},
+				args:   []string{"course.yaml"},
 				runAll: true,
 			},
-			want:    "testdata/course.yaml",
-			wantErr: false,
+			want:    "",
+			wantErr: true,
 		},
 		{
-			name: "course.yaml exists, length of args = 2",
+			name: "course.yaml exists, pass onlyrun with success",
 			args: args{
-				args:    []string{"testdata/course.yaml", "course.yml"},
+				args:    []string{"testdata/course.yaml"},
 				runAll:  false,
 				onlyRun: []string{"rbac-manager"},
 			},
-			wantErr: true,
+			want:    "testdata/course.yaml",
+			wantErr: false,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, err := ValidateArgs(tt.args.runAll, tt.args.onlyRun, tt.args.args)
+			// note: I do not like repeating logic from cmd/root.go in
+			// this test. instead, or additionally, perhaps we should
+			// test root.go directly to gain its effects
```

Comment:
I don't know if you can get around that unless you move that logic all into this package and test it fully here. Even if you test root.go you'll need to get a course.yml to work with in this test.

Also, lets remove this comment block.

Reply:
Thinking about this, it seems to be it would be easier to just move these validating functions into the cmd package where they really matter. I'll add that to my top level review comment.

Reply:
Moved code & removed comment.